### ISSUE TITLE
Logging feature?

### ISSUE THREAD

**Issue by nicolas-heigvd** [opengeospatial/ogcapi-processes#273]
Short: as a developer of processing (geo-)routines, I wish I could share with the clients some logs through specific endpoint(s) of the OGC API - Processes in order to share these important pieces of information with them.

This is maybe a naive question but I cannot figure out an answer in the draft standard document: is the "OGC API - Processes" featuring some logging abilities or ability to collect and return some processing logs, _separate_ from the results per se, e.g. for sharing intermediate states, information on particular processing sub-steps, errors, timestamps, etc., i.e, whatever they may be (without having to specify for each individual log feature a specific optional property)?

**Comment by pvretano** [comment]
@nicolas-heigvd this is not specifically covered in the specification since it is more of an implementation detail BUT you can provide a link to a run log in the response's link section and/or the HTTP header.  My server does that and I use the link relation `rel="runlog"` to identify the relevant link.  Here is an example I mocked up:

```
{
  "jobID" : "81574318-1eb1-4d7c-af61-4b3fbcf33c4f",
  "status": "completed",
  "message": "Process completed successfully.",
  "progress": 100,
  "created": "2021-05-04T10:13:00+05:00",
  "links": [
    ...
    {
      "href": "... some opaque link to the run log for this job ...",
      "rel": "runlog",
      "type": "application/text"
    }
    ...
  ]
}
```

**Comment by fmigneault** [comment]
Currently, I use the following: 
`/jobs/{job_id}/logs` (JSON)
`/jobs/{job_id}/logs?f=text` (or using `Accept: text/plain`)
https://pavics-weaver.readthedocs.io/en/latest/api.html#tag/Logs
And the Job Status provides `rel: logs` with a link to the above for the specific job.

Should this be added as a recommendation? 
If an OGC NA is defined explicitly for logs, it should make it clear whether a specific media-type is expected or not.

**Comment by pvretano** [comment]
The CubeWerx server uses the endpoint `/jobs/{jobID}/log` (singular) to get the run log and the link relation is "runlog".  By default, the output format is "text/plain" but that can be negotiated to something else like JSON.  

I am not opposed to it but I don't know if we need to define an actual endpoint for getting the run logs.  Isn't having a link in the job status sufficient?

I checked the name server and I don't see a "rel" specifically defined for logs.  Mind you, navigating the OGC Name Server is a bit of a trick so I could have missed it.  If there isn't a suitable rel then I would propose that we use something like: `http://www.opengis.net/def/rel/ogc/1.0/runlog` which could be shorten using a CURIE to `[ogc:runlog]`.  I prefer the name "runlog" because, in this context, it is more specific then the generic term "log".  

I am not particularly religious about whether we define an endpoint, the use of singular "log" vs "logs" or the name of the rel ... just stating my thinking.

**Comment by fmigneault** [comment]
I agree about not necessarily having an explicit endpoint. Some servers could simply point at some AWS bucket where the logs could be located for example, so the link is sufficient. I'm fine with that `rel` name if it gets registered.
